prototype/galore/galore_fused/triton/custom_autotune.py

```diff
@@ -0,0 +1,392 @@
+from __future__ import annotations
+
+import builtins
+import logging
+import os
+import time
+from typing import Dict
+
+import numpy as np
+from triton.runtime.cache import default_cache_dir
+from triton.runtime.errors import OutOfResources
+from triton.runtime.jit import KernelInterface
+from triton.testing import do_bench
+
+logger = logging.getLogger(__file__)
+
+
+class Autotuner(KernelInterface):
```

Comment:
@cpuhrsch we have a generic kernel auto tuner now right?

Reply:
I tweaked the `Autotuner` to print additional info such as pruned configs, best config, cache hit, etc.